tests/models/mllama/test_processor_mllama.py

```diff
@@ -13,32 +13,44 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import shutil
+import tempfile
 import unittest
+from typing import Optional
 
 import numpy as np
 
 from transformers import MllamaProcessor
 from transformers.testing_utils import require_torch, require_vision
 from transformers.utils import is_vision_available
 
+from ...test_processing_common import ProcessorTesterMixin
+
 
 if is_vision_available():
     from PIL import Image
 
 
 @require_torch
 @require_vision
-class MllamaProcessorTest(unittest.TestCase):
+class MllamaProcessorTest(ProcessorTesterMixin, unittest.TestCase):
+    processor_class = MllamaProcessor
+
     def setUp(self):
         self.checkpoint = "hf-internal-testing/mllama-11b"  # TODO: change
-        self.processor = MllamaProcessor.from_pretrained(self.checkpoint)
+        processor = MllamaProcessor.from_pretrained(self.checkpoint)
         self.image1 = Image.new("RGB", (224, 220))
         self.image2 = Image.new("RGB", (512, 128))
-        self.image_token = self.processor.image_token
-        self.image_token_id = self.processor.image_token_id
-        self.pad_token_id = self.processor.tokenizer.pad_token_id
-        self.bos_token = self.processor.bos_token
-        self.bos_token_id = self.processor.tokenizer.bos_token_id
+        self.image_token = processor.image_token
+        self.image_token_id = processor.image_token_id
+        self.pad_token_id = processor.tokenizer.pad_token_id
+        self.bos_token = processor.bos_token
+        self.bos_token_id = processor.tokenizer.bos_token_id
+        self.tmpdirname = tempfile.mkdtemp()
+        processor.save_pretrained(self.tmpdirname)
```

Comment:
changed the use of `self.processor` to this to have a "clean" processor at the beginning of each test.

Reply:
That's a good idea, could be replaced by a fixture with a `function` scope then?

Reply:
> That's a good idea, could be replaced by a fixture with a function scope then?
Actually my bad, it looks like the `setUp` and `tearDown` functions are already fixtures with a function scope for `unittest`. So it's not necessary to use tempfile.
I don't know then why saving processors to a tempfile is used in the setUp function of so many test_processor files, and why tempfile in general is used in so many unittest setUp function in Transformers.
Unless there is a good reason to use tempfiles, maybe we should think about removing its use in setUp functions in all tests?

Reply:
`self.tmpdirname` is used in ProcessorTesterMixin so I can't remove it here in fact, but maybe we could consider removing it from ProcessorTesterMixin in another PR

Reply:
No strong opinion on it - it's just convenient to use and makes sure the save/load from files works, can't create race conditions either. Since so many of the transformers utils are designed to work in pair with the hub, I suppose it makes sense to use `tempfile` in that regard

Reply:
Oh I see, I have no strong opinion on it either, so maybe no need to change it :)